scripts/stav-covid.py

```diff
@@ -0,0 +1,167 @@
+output = requests.get(url).text
+
+# Now, parse it.
+output = BeautifulSoup(output, features="html.parser")
+
+# NOTE(rye): There is no filtering to prevent loading tables that aren't valid,
+# so please double-check the output of this script.
+tables = output.find_all("table")
+
+# A regex to match strings like "Sunday, August 23"
+date_re = re.compile('\w+, \w+ \d{1,2}')
+
+# A regex used to match timespec delimiters (e.g. the "-"" in "9-10:45")
+timespec_re = re.compile('\s*-\s*|\s*–\s*')
+
+# If the given tag only has one set of contents, just grab the contents.
+def meal_name_from(th):
+	if len(th.contents) == 1:
+		return th.get_text()
+	else:
+		return None
+
+# Attempt to parse a lazy timestring (either hh or hh:mm) to a datetime.  Note
+# that this datetime is context-free, and needs to be adjusted to make sense.
+def timeparse(string, date=None):
+	parts = string.split(":")
+	if len(parts) == 1:
+		return datetime.datetime.combine(date, datetime.time(hour=int(parts[0])))
+	elif len(parts) == 2:
+		return datetime.datetime.combine(date, datetime.time(hour=int(parts[0]), minute=int(parts[1])))
+	else:
+		return None
+
+def schedules_to_document(schedules):
+	name = "Stav Hall — {}".format(dorm)
+	document = { "name": name, "image": "stav", "category": "Food", "schedule": list(schedules.values()) }
+	return document
+
+def render_schedule_to_file(schedules, fmt, dorm="Unknown"):
+	# TODO something something globals
+	file = fmt.format(re.sub("\s", "-", dorm.lower().strip()))
+
+	print("Rendering schedule for dorm {} to file {}".format(dorm, file))
+
+	with open(file, "w") as file:
+		yaml.dump(schedules_to_document(schedules), file, default_flow_style=None)
+
+# Start with an empty schedule set
+schedules = {}
+
+# For each of the HTML tables we found,
+for table in tables:
+	print("Processing a table...")
+
+	# Find the prior sibling, which is an h3 containing the day in some format.
+	# Each table should have an h3 before it that has date information in it.
+	header = table.find_previous_sibling("h3").get_text().strip()
+	print('Found a header sibling with text "{}"'.format(header))
+
+	# Search the text for just the date
+	matching_text = date_re.match(header).group(0)
+
+	# NOTE(rye): Parses text like "Sunday, August 23"
+	date = datetime.datetime.strptime(matching_text, "%A, %B %d")
+
+	# Move the date to the current year
+	date = datetime.datetime(date.today().year, date.month, date.day)
+
+	# If the computed date is more than six months before today, assume it actually belongs in the next year.
+	if date < date.today() + datetime.timedelta(days=-180):
+		date = datetime.datetime(date.year + 1, date.month, date.day)
+
+	# We only know the date, so let's back out to that.
+	date = date.date()
+
+	print("Inferred date of {}; processing table now.".format(date))
+
+	body = table.find("tbody")
+
+	# NOTE(rye): Should only have one thead
+	meals = [meal_name_from(th) for th in table.find("thead").find("tr").find_all("th")]
```

Comment:
Do we want to `assert len(meals) == 1`, then?

Reply:
If I cared much about error handling, certainly—that'd require doing a separate search for `table.find_all("thead")` instead…